### PR TITLE
Implement PR check with test coverage info colection for SonarQube

### PR DESCRIPTION
**Pull request by Rominos7**
### JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

### About this PR

This PR add PR-check for Street Code client repo along with test coverage collection for Sonar Cloud

Link to sonar with this PR analys
https://sonarcloud.io/summary/new_code?id=ita-social-projects_StreetCode_Client&pullRequest=1178

Other usefull links:
https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/analysis-parameters/#project-information
https://github.com/SonarSource/sonarqube-scan-action

https://docs.sonarsource.com/sonarcloud/advanced-setup/automatic-analysis/#conflict-with-ci-based-analysis
https://community.sonarsource.com/t/coverage-with-automatic-analysis-in-sonarcloud/29318

### Check list
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  PR meets all conventions

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    # when lint error will be fixed - add step with lint as well
+
+    - name: Run unit tests
+      run: npm run test:cover
+    
+    - name: SonarQube Scan
+      uses: sonarsource/sonarqube-scan-action@master
+      env:
+        SONAR_TOKEN: a81a0f0e8452e520a1067e4eed2669d10f059b7a
```
Comment (Rominos7) [inline .github/workflows/build.yml:32]: Consider to move this token to the env variables

**Review by Adeodonne — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (LanchevychMaxym) [reply .github/workflows/build.yml:32]: You can use GitHub secrets for now

**Review by LanchevychMaxym — COMMENTED** [review]
LGTM!

**Review by LanchevychMaxym — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Rominos7) [reply .github/workflows/build.yml]: Done

**Review by IvannYad — APPROVED** [review]

**Pull request merged**